test/common/common/thread_id_test.cc

```diff
@@ -0,0 +1,49 @@
+#include "common/common/thread.h"
+#include "test/test_common/thread_factory_for_test.h"
+
+#include "gmock/gmock.h"
+#include "absl/hash/hash_testing.h"
+
+namespace Envoy {
+namespace {
+
+TEST(ThreadId, Equality) {
+  auto& thread_factory = Thread::threadFactoryForTest();
+
+  Thread::ThreadId main_thread = thread_factory.currentThreadId();
+  Thread::ThreadId background_thread;
+  Thread::ThreadId null_thread;
+
+  Thread::threadFactoryForTest()
+      .createThread([&]() { background_thread = thread_factory.currentThreadId(); })
+      ->join();
+
+  EXPECT_EQ(main_thread, main_thread);
+  EXPECT_EQ(background_thread, background_thread);
+  EXPECT_EQ(null_thread, null_thread);
+
+  EXPECT_NE(main_thread, background_thread);
+  EXPECT_NE(main_thread, null_thread);
+  EXPECT_NE(background_thread, null_thread);
+}
+
+TEST(ThreadId, Hashability) {
+  auto& thread_factory = Thread::threadFactoryForTest();
+
+  Thread::ThreadId main_thread = thread_factory.currentThreadId();
+  Thread::ThreadId background_thread;
+  Thread::ThreadId null_thread;
+
+  Thread::threadFactoryForTest()
+      .createThread([&]() { background_thread = thread_factory.currentThreadId(); })
+      ->join();
+
+  EXPECT_TRUE(absl::VerifyTypeImplementsAbslHashCorrectly({
```

Comment:
nice tests; did not know about this test-helper function. Pretty cool. I do sometimes use tests as doc though to understand how to instantiate. I assume we can just directly instantiate flat_hash_set<ThreadId> without worry?

Reply:
yup